cpp/include/raft/core/resource/nccl_clique.hpp

```diff
@@ -51,14 +79,75 @@ class nccl_clique_resource_factory : public resource_factory {
  * @param[in] res the raft resources object
  * @return NCCL clique
  */
-inline const raft::comms::nccl_clique& get_nccl_clique(resources const& res)
+inline std::vector<raft::resources>& get_nccl_clique(resources const& res)
 {
   if (!res.has_resource_factory(resource_type::NCCL_CLIQUE)) {
     res.add_resource_factory(std::make_shared<nccl_clique_resource_factory>());
   }
-  return *res.get_resource<raft::comms::nccl_clique>(resource_type::NCCL_CLIQUE);
+  return *res.get_resource<std::vector<raft::resources>>(resource_type::NCCL_CLIQUE);
+};
+
+/**
+ * @brief Get number of ranks in clique
+ */
+inline int get_num_ranks(resources const& res)
+{
+  return raft::resource::get_nccl_clique(res).size();
+}
+
+/**
+ * @brief Get rank's raft::resources object
+ */
+inline const raft::resources& get_device_resources(resources const& res, int rank)
```

Comment:
Just to make this more generally clear, can we call this `get_device_resources_for_rank`?